pkg/internal/cmd/options/options.go

```diff
@@ -17,47 +17,123 @@ limitations under the License.
+
+	// LeaderElectionNamespace is the namespace in which leader election should
+	// be leased in to form leader election.
 	LeaderElectionNamespace string
 
-	zapOptions *zap.Options
-	Log        logr.Logger
+	// RestConfig is the shared based rest config to connect to the Kubernetes
+	// API.
+	RestConfig *rest.Config
+
+	// Logr is the shared base logger.
+	Logr logr.Logger
 }
 
 func New() *Options {
 	return new(Options)
 }
 
-func (o *Options) AddFlags(cmd *cobra.Command) *Options {
-	fs := flag.NewFlagSet("policy-approver", flag.ExitOnError)
+func (o *Options) Prepare(cmd *cobra.Command) *Options {
+	o.addFlags(cmd)
+	return o
+}
+
+func (o *Options) Complete() error {
+	klog.InitFlags(nil)
+	log := klogr.New()
+	flag.Set("v", o.logLevel)
+	o.Logr = log
 
-	o.zapOptions = &zap.Options{
-		Development: false,
+	var err error
+	o.RestConfig, err = o.kubeConfigFlags.ToRESTConfig()
+	if err != nil {
+		return fmt.Errorf("failed to build kubernetes rest config: %s", err)
 	}
-	o.zapOptions.BindFlags(fs)
 
-	fs.StringVar(&o.MetricsAddress, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	fs.StringVar(&o.ProbeAddress, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	fs.StringVar(&o.LeaderElectionNamespace, "leader-election-namespace", "cert-manager", "leader election namespace to use for the controller manager")
+	return nil
+}
 
-	cmd.PersistentFlags().AddGoFlagSet(fs)
+func (o *Options) addFlags(cmd *cobra.Command) {
+	var nfs cliflag.NamedFlagSets
 
-	return o
+	o.addAppFlags(nfs.FlagSet("App"))
+	o.kubeConfigFlags = genericclioptions.NewConfigFlags(true)
+	o.kubeConfigFlags.AddFlags(nfs.FlagSet("Kubernetes"))
+
+	usageFmt := "Usage:\n  %s\n"
+	cmd.SetUsageFunc(func(cmd *cobra.Command) error {
+		fmt.Fprintf(cmd.OutOrStderr(), usageFmt, cmd.UseLine())
+		cliflag.PrintSections(cmd.OutOrStderr(), nfs, 0)
+		return nil
+	})
+
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		fmt.Fprintf(cmd.OutOrStdout(), "%s\n\n"+usageFmt, cmd.Long, cmd.UseLine())
+		cliflag.PrintSections(cmd.OutOrStdout(), nfs, 0)
+	})
+
+	fs := cmd.Flags()
+	for _, f := range nfs.FlagSets {
+		fs.AddFlagSet(f)
+	}
 }
 
-func (o *Options) Complete() {
-	o.Log = zap.New(zap.UseFlagOptions(o.zapOptions))
+func (o *Options) addAppFlags(fs *pflag.FlagSet) {
+	fs.StringVarP(&o.logLevel, "log-level", "v", "1",
+		"Log level (1-5).")
+
+	fs.StringVar(&o.MetricsAddress, "metrics-bind-address", ":9402",
+		`TCP address for exposing HTTP Prometheus metrics which will be served on the HTTP path '/metrics'. The value "0" will
+	 disable exposing metrics.`)
+
+	fs.StringVar(&o.ReadyzAddress, "readiness-probe-bind-address", ":6060",
+		"TCP address for exposing the HTTP readiness probe which will be served on the HTTP path '/readyz'.")
+
+	fs.BoolVar(&o.ApproveWhenNoPolicies, "approve-when-no-policies", false,
+		"TCP address for exposing the HTTP readiness probe which will be served on the HTTP path '/readyz'.")
```

Comment:
Nit: I think this description is for a different flag.

Reply:
Thanks :smile: 

As mentioned, this flag is going away.